### PR TITLE
Add initial membership documentation

### PR DESCRIPTION
**Pull request by cpanato**
#### Summary
- Pick the commit from kubernetes to give some attribution for the already existing documentation https://github.com/kubernetes/community/commit/f1222b4a3323c082bb8a2aaa1d481a002bb53aaa
- Adding initial membership documentation that we can iterate over to refine and make that align with Sigstore organization.

Need to add some documentation regarding the GitHub org, which I will push  in a follow-up PR

/cc @lukehinds @bobcallaway @dlorenc 

#### Ticket Link
xref: https://github.com/sigstore/sigstore/issues/305 / https://github.com/sigstore/community/issues/53
#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: OWNERS.md [hunk 1]
```diff
@@ -1,4 +1,5 @@
 # OWNERS (TSC Members)
+
 * Luke Hinds (**[@lukehinds](https://github.com/lukehinds)**), Red Hat
 * Bob Callaway (**[@bobcallaway](https://github.com/bobcallaway)**), Red Hat
```
Comment (bobcallaway) [inline OWNERS.md:4]: Could you update this for me to `Google` please? :)

Comment (cpanato) [reply OWNERS.md:4]: ah yeah! that is true forgot that

**Comment by cpanato** [comment]
when have some free cycle i would love to get some feedback @dlorenc @lukehinds

**Comment by dlorenc** [comment]
This looks like a great start to me! No concerns.

**Comment by lukehinds** [comment]
Sorry I missed this late time around. In regards to folks joining the org, we are right now maxed out and cannot take anymore on due to githubs default limits.

File: community-membership.md [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+This doc outlines the various responsibilities of contributor roles in
+Sigstore.
+Responsibilities for most roles are scoped to these subprojects.
+
+| Role | Responsibilities | Requirements | Defined by |
+| -----| ---------------- | ------------ | -------|
+| Member | Active contributor in the community | Sponsored by 2 reviewers and multiple contributions to the project | Sigstore GitHub org member|
+| Reviewer | Review contributions from other members | History of review and authorship in a subproject | [CODEOWNERS] file reviewer entry |
+| Approver | Contributions acceptance approval| Highly experienced active reviewer and contributor to a subproject | [CODEOWNERS] file approver entry|
```
[the remark this answers is not in the document]
Comment (haydentherapper) [reply community-membership.md:13]: Just curious, is there a benefit to using the team membership over using CODEOWNERS? The benefit of the latter is we can create per-file/per-directory ownership.

File: community-membership.md [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+- Code reviewer status may be a precondition to accepting large code contributions
+- Responsible for project quality control via [code reviews]
+  - Focus on code quality and correctness, including testing and factoring
+  - May also review for more holistic issues, but not a requirement
+- Expected to be responsive to review requests as per [community expectations]
+- Assigned PRs to review related to the project of expertise
+- Assigned test bugs related to the project of expertise
+- May get a badge on PR and issue comments
+
+## Approver
```
Comment (haydentherapper) [inline community-membership.md:117]: Is there a reason we're splitting the two of these? All of the things you mention in approver, understanding API conventions, performance, interactions, etc, I think should be the responsibility of the reviewer role also.

There isn't a functional difference between the two in CODEOWNERS, right? I see a mention here that approvers are top-level, is the thought that reviewers would be per-directory?

My thought is these should be the same role in the community guidelines. You either can be a top-level codeowner if you have demonstrated an understanding of the entire project, or a per-directory/package codeowner if you have an understanding of a smaller part of the codebase.

File: community-membership.md [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+
+### Requirements
+
+The following apply to the part of codebase for which one would be a reviewer in
+an [CODEOWNERS] file (for repos using the bot).
+
+- member for at least 3 months
+- Primary reviewer for at least 5 PRs to the codebase
+- Reviewed or merged at least 20 substantial PRs to the codebase
+- Knowledgeable about the codebase
```
Comment (haydentherapper) [inline community-membership.md:96]: I'd also add being a subject matter expert in the area of the code that they're responsible for, i.e PKI, transparency logs, OCI, etc.

File: community-membership.md [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+addition to the assigned reviewers.
+
+### Requirements
+
+The following apply to the part of codebase for which one would be a reviewer in
+an [CODEOWNERS] file (for repos using the bot).
+
+- member for at least 3 months
+- Primary reviewer for at least 5 PRs to the codebase
+- Reviewed or merged at least 20 substantial PRs to the codebase
```
Comment (haydentherapper) [inline community-membership.md:95]: Maybe also a substantial number of features? 20/30 PRs seems pretty arbitrary to me. I assume the purpose is to make sure the maintainer is knowledgeable about the codebase, which can be done through feature development.

File: community-membership.md [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+- Knowledgeable about the codebase
+- Sponsored by a project approver
+  - With no objections from other approvers
+  - Done through PR to update the OWNERS file
```
Comment (haydentherapper) [inline community-membership.md:99]: nit: CODEOWNERS

File: community-membership.md [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+- May either self-nominate, be nominated by an approver in this subproject, or be nominated by a robot
```
Comment (haydentherapper) [inline community-membership.md:100]: Nominated by a robot? Do we want automation for this?

**Comment by naveensrinivasan** [comment]
What is the status of this PR? This can help the other OpenSSF communities trying to do something similar.

**Comment by cpanato** [comment]
> What is the status of this PR? This can help the other OpenSSF communities trying to do something similar.

i will update and get back to this early next week

**Comment by haydentherapper** [comment]
@cpanato Would you like any help with the updates? Happy to collaborate.

File: community-membership.md [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+
+## Inactive members
+
+_Members are continuously active contributors in the community._
+
+A core principle in maintaining a healthy community is encouraging active
+participation. It is inevitable that people's focuses will change over time and
+they are not expected to be actively contributing forever.
+
+_TBD how to track inactive members and phase down._
```
Comment (haydentherapper) [inline community-membership.md:164]: Maybe some guidelines:

* Lack of significant contribution to the repository
* Not assisting with reviews or triaging issues

File: community-membership.md (hunk 5 above)
Comment (cpanato) [reply community-membership.md:95]: makes sense

File: community-membership.md (hunk 7 above)
Comment (cpanato) [reply community-membership.md:100]: clean up

File: community-membership.md (hunk 3 above)
Comment (cpanato) [reply community-membership.md:117]: agree, with you

Comment (cpanato) [reply community-membership.md:117]: the reviewer in our case can be more the triage role, will update that

**Comment by cpanato** [comment]
@haydentherapper @naveensrinivasan @bobcallaway did some updates, can I get another review round and let me know what is missing or need to update.
You can also push changes to the PR if you want :)

thanks so much for all the feedback

**Comment by dlorenc** [comment]
This seems like a good start to me!

**Review by bobcallaway — DISMISSED** [review]
lgtm as well

File: .github/ISSUE_TEMPLATE/membership.yml [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+  attributes:
+    label: "Sponsor 2"
+    description: GitHub handle of your sponsor
+    placeholder: e.g. @sponsor-2
+  validations:
+    required: true
+- id: contributions
+  type: textarea
+  attributes:
+    label: List of contributions to the Sigstgore project
```
Comment (lukehinds) [inline .github/ISSUE_TEMPLATE/membership.yml:61]: Sigstgore / Sigstore

**Review by lukehinds — COMMENTED** [review]
Just a typo, but other than that lgtm

File: .github/ISSUE_TEMPLATE/membership.yml (hunk 9 above)
Comment (cpanato) [reply .github/ISSUE_TEMPLATE/membership.yml]: thanks! fixed

**Comment by cpanato** [comment]
need another lgtm after the rebase with the typo fix

**Review by bobcallaway — APPROVED** [review]

**Pull request merged**